base/acme/src/main/java/org/dogtagpki/acme/server/ACMEEngine.java

```diff
@@ -817,37 +817,43 @@ public void addOrder(ACMEAccount account, ACMEOrder order) throws Exception {
         database.addOrder(order);
     }
 
-    enum CheckOrderResult { OrderAccountMismatch , OrderExpired , OrderAccessOK };
+    enum CheckOrderResult { ORDER_ACCOUNT_MISMATCH , ORDER_EXPIRED , ORDER_ACCESS_OK, ORDER_NULL}
 
     public CheckOrderResult checkOrder(ACMEAccount account, ACMEOrder order) {
 
-        String orderID = order.getID();
+        if (order == null) {
+            return CheckOrderResult.ORDER_NULL;
+        }
+
         if (!order.getAccountID().equals(account.getID())) {
-            return CheckOrderResult.OrderAccountMismatch;
+            return CheckOrderResult.ORDER_ACCOUNT_MISMATCH;
         }
 
         Date expirationTime = order.getExpirationTime();
         if (expirationTime == null) {
-            return CheckOrderResult.OrderAccessOK;
+            return CheckOrderResult.ORDER_ACCESS_OK;
         }
 
         long currentTime = System.currentTimeMillis();
         if (expirationTime.getTime() <= currentTime) {
-            return CheckOrderResult.OrderExpired;
+            return CheckOrderResult.ORDER_EXPIRED;
         }
 
-        return CheckOrderResult.OrderAccessOK;
+        return CheckOrderResult.ORDER_ACCESS_OK;
     }
 
     public void validateOrder(ACMEAccount account, ACMEOrder order) throws Exception {
         switch (checkOrder(account, order)) {
-            case OrderAccountMismatch:
+            case ORDER_ACCOUNT_MISMATCH:
                 // TODO: generate proper exception
                 throw new Exception("Unable to access order " + order.getID());
-            case OrderExpired:
+            case ORDER_EXPIRED:
                 // TODO: generate proper exception
                 throw new Exception("Expired order: " + order.getID());
-            case OrderAccessOK:
+            case ORDER_NULL:
+                // TODO: generate proper exception
+                throw new Exception("Order is null");
```

Comment:
Could we say "Order is missing" or something like that? That might be more user-friendly.

Reply:
Yeah probably better, ack.